tests/vault_action_tests_base.py

```diff
@@ -0,0 +1,81 @@
+from st2tests.base import BaseActionTestCase
+
+from tests.utils import get_config_file_path
+from tests.utils.hvac_integration_test_case import HvacIntegrationTestCase
+
+
+class VaultActionTestCase(HvacIntegrationTestCase, BaseActionTestCase):
+    dummy_pack_config = None
+    secret_v1 = False
+    secret_v2 = False
+    default_token_lease = "1h"
+
+    # NB: self.client has a root_token. Use for test setup, but not for action tests.
+
+    def setUp(self):
+        # NOTE: HvacIntegrationTestCase.setUp() mocks hvac.utils.warnings
+        # which means that deprecations warnings will be swallowed in CI.
+        # We might need to work around that at some point.
+
+        super(VaultActionTestCase, self).setUp()
+        # HvacIntegrationTestCase does not call super(), so we take care of that.
+        super(HvacIntegrationTestCase, self).setUp()
+
+        self.dummy_pack_config = self.build_dummy_pack_config()
+
+        mounted_secrets_engines = self.client.sys.list_mounted_secrets_engines()["data"]
+        # based on hvac/tests/integration_tests/v1/test_integration.py
+        if "secret/" not in mounted_secrets_engines:
+            self.client.enable_secret_backend(
+                backend_type="kv",
+                mount_point="secret",
+                options=dict(version=1),
+            )
+
+        # based on hvac/tests/integration_tests/api/secrets_engins/test_kv_v*.py
+        if self.secret_v1 and "kvv1/" not in mounted_secrets_engines:
+            self.client.enable_secret_backend(
+                backend_type="kv",
+                mount_point="kvv1",
+                options=dict(version=1),
+            )
+        if self.secret_v2 and "kvv2/" not in mounted_secrets_engines:
+            self.client.enable_secret_backend(
+                backend_type="kv",
+                mount_point="kvv2",
+                options=dict(version=2),
+            )
+
+    def tearDown(self):
+        super(VaultActionTestCase, self).setUp()
+        # HvacIntegrationTestCase does not call super(), so we take care of that.
+        super(HvacIntegrationTestCase, self).setUp()
```

Comment:
NAK - this calls `setUp()` again, not `tearDown()` like it should.

```suggestion
        super(VaultActionTestCase, self).tearDown()
        # HvacIntegrationTestCase does not call super(), so we take care of that.
        super(HvacIntegrationTestCase, self).tearDown()
```